### DISCUSSION TITLE
Does mp3 play works without external DAC on ESP32?

### DISCUSSION THREAD

**Discussion by Mir1001** [earlephilhower/ESP8266Audio#699] [Q&A]
I got board with amplifier ICs FM8002A and ESP32. I am able to play mp3, but sound is heavy distorted (Wav and mp3 file). If I play artificially generated sound it almost works. I have tried different combinations:

Best I got [listen here](https://drive.google.com/file/d/1wSgHgIH751F18XzHLgn9X70YDHqvznZX/view)
```cpp
AudioOutputI2S *outs(int o) //o - selected option
{
    switch (o)
    {
    case 0: // distorted almost works
        return new AudioOutputI2SNoDAC();
    case 1:
        return new AudioOutputI2S(); // loud and distorted
    case 2:                          // sample rate is too small, the mclk division exceed the maximum value 255
        return new AudioOutputI2S(0, AudioOutputI2S::INTERNAL_DAC);
    case 3: // i2s_calculate_pdm_tx_clock(720): sample rate is too small, the mclk division exceed the maximum value 255
        return new AudioOutputI2S(0, AudioOutputI2S::INTERNAL_PDM, 8, AudioOutputI2S::APLL_ENABLE);
    case 4: // loud and distorted
        return new AudioOutputI2S(0, AudioOutputI2S::EXTERNAL_I2S, 8, AudioOutputI2S::APLL_ENABLE);
    case 5:  // loud and distorted
        return new AudioOutputI2S(0, AudioOutputI2S::EXTERNAL_I2S, 8, AudioOutputI2S::APLL_DISABLE);
    default:
        return nullptr; // Handle unexpected values of 'o'
    }
};
```

Can someone give me any suggestion at lest confirm that it should work or that you had same problems ;(.

Thanks

**Comment by anupam-dubey** [comment]
i stuck with the same issue

#include "AudioFileSourceSPIFFS.h"
#include "AudioGeneratorMP3.h"
#include "AudioOutputI2S.h"
#include "SPIFFS.h"

AudioGeneratorMP3 *mp3;
AudioFileSourceSPIFFS *file;
// The library can be configured to use the internal DAC via I2S
AudioOutputI2S *out; 

void setup() {
  Serial.begin(115200);
  SPIFFS.begin(true); // Start SPIFFS
  Serial.println("SPIFFS mounted");

  // Specify the MP3 file path
  file = new AudioFileSourceSPIFFS("/pno-cs.mp3"); 

  // Initialize AudioOutputI2S, specifying the DAC pins. 
  // You may need to check the library documentation for specific 
  // configuration to route to the internal DAC pins (GPIO 25/26) directly.
  // The snippet below shows a generic setup, often configured 
  // via setPinout for I2S modules, but for the internal DAC, the 
  // library will manage the correct output.
  out = new AudioOutputI2S(0, false); // Example, check library specifics for direct DAC pin usage
//  out->setPins(GPIO_NONE, GPIO_NONE, GPIO_NONE); // Ensure I2S pins are not set if using internal DAC directly or configure for DAC output

  mp3 = new AudioGeneratorMP3();
  mp3->begin(file, out);
}

void loop() {
  if (mp3->isRunning()) {
    if (!mp3->loop()) {
      mp3->stop();
      Serial.println("MP3 playback finished");
    }
  } else {
    // Optional: add code here to restart playback or stop
  }
}

**Comment by anupam-dubey** [comment]
not provding any audio to be cascaded with amplifier
